### ISSUE TITLE
Swipe to dismiss gesture shows black part of screen before full dismissal

### ISSUE THREAD

**Issue by abdulla-allaith-7** [jonkykong/SideMenu#213]
## New Issue Checklist

I have read the [guidelines for contributing](https://github.com/jonkykong/SideMenu/blob/master/.github/CONTRIBUTING.md) and I understand:
- [x] My issue is happening in the **latest version** of SideMenu.
- [x] My issue was **not** solved in the [README](https://github.com/jonkykong/SideMenu/blob/master/README.md).
- [x] My issue can **not** be answered on [stackoverflow.com](stackoverflow.com).
- [x] My issue is **not** a request for new functionality that I am unwilling to build and contribute with a pull request.
- [x] My issue **is** reproducible in the [demo project](https://github.com/jonkykong/SideMenu/tree/master/Example).

## Issue Description

There's a problem when dismissing the side menu with using gesture. The issue does not occur when dismissing using a button.
<br>
The problem happens specifically when you swipe to dismiss the menu, but do not drag it all the way to the end. When you release the menu near enough for it to be closed, the menu turns to black as shown in this image:
<br>
<img height=550 width=300 src="https://user-images.githubusercontent.com/29616360/27412119-175e2d68-56fb-11e7-82d2-5a7a0072c1ae.png">
<br>
The issue as shown in the above image happens when using `Slide Out` presenting mode, and it also happens -**even more apparent**- with `In + Out` presenting mode!
<br>
I have tried to inspect what is happening and it does not seem to be related only to the background color turning black, as the whole view is empty & black. It seems as if the `SideMenuTransition` is using `view.removeFromSuperview()` sooner than it should, or another problem somewhere.

Please fix this, as it is the only issue I face with an otherwise perfect slide menu!

**Comment by jonkykong** [comment]
@abdulla-allaith-7 you are the only person to report this problem so far, and I am unable to reproduce this myself in the demo project. What version of iOS are you testing on?

**Comment by abdulla-allaith-7** [comment]
@jonkykong iOS 10, Xcode 8.3.3.          
I'm using cocoapods for my project, but I tried the demo and it's also there.        
Swiping the menu (not until the very end) and causing it to close shows black menu for the remaining of it's hide animation. If you are still unable to reproduce this, I can post a GIF showing exactly how it's produced.

**Comment by jonkykong** [comment]
@abdulla-allaith-7 based on your current explanation, I'm not reproducing it. It behaves exactly as intended.

I reveal the menu, then release my finger. Then I start dragging back to dismiss. While the menu is still visible, I release it, and it finishes. I then swipe to reveal again without issues. Is this reproducing in the simulator for you as well?

**Comment by jonkykong** [comment]
Closing for now without additional information or follow-up reports. Will re-open as needed.
